code/modules/projectiles/ammunition/energy/special.dm

```diff
@@ -65,3 +65,40 @@
 	e_cost = 200
 	select_name = "stun"
 	projectile_type = /obj/item/projectile/energy/tesla/revolver
+
+// HoS ballistic gun settings
+
+/obj/item/ammo_casing/energy/hos
+	fire_sound = 'sound/weapons/revolver357shot.ogg'
+	name = "integrated miniature 3D printer"
+	desc = "A miniaturised 3D printer, capable of running off an energy gun cell to produce .454 bullets for immediate use."
+	select_name = "boolet"
+	projectile_type = /obj/item/projectile/bullet/hos
+
+/obj/item/ammo_casing/energy/hos/hv
+	e_cost = 120
+	select_name = ".454HV"
+	projectile_type = /obj/item/projectile/bullet/hos/hv
+
+/obj/item/ammo_casing/energy/hos/heavy
+	e_cost = 200
+	select_name = ".454HI"
+	projectile_type = /obj/item/projectile/bullet/hos/heavy
+
+/obj/item/ammo_casing/energy/hos/light
+	e_cost = 60
+	select_name = ".454AR"

```

Comment:
With the 3 pellet bit, and boosted fire rate, a cost increase here is justified

Reply:
Done.